src/__test__/stellar-service.spec.ts

```
@@ -1,45 +1,321 @@
-      const validation = blindpay.stellar.validatePublicKey(publicKey);
-      expect(validation).toBe(true);
+  describe('Mint Asset', () => {
+    let issuerPublicKey: string;
+    let issuerSecretKey: string;
+    let receiverPublicKey: string;
+    let receiverSecretKey: string;
+    const assetCode = 'USDB';
+
+    beforeEach(async () => {
+      const issuerKeypair = Keypair.random();
+      const receiverKeypair = Keypair.random();
+      await server.friendbot(issuerKeypair.publicKey()).call();
+      await server.friendbot(receiverKeypair.publicKey()).call();
+
+      issuerPublicKey = issuerKeypair.publicKey();
+      issuerSecretKey = issuerKeypair.secret();
+      receiverPublicKey = receiverKeypair.publicKey();
+      receiverSecretKey = receiverKeypair.secret();
+    });
+
+    describe('Returned Response', () => {
+      it('Should return a transaction hash if the asset is minted', async () => {
+        const response = await blindpay.stellar.mintAsset({
+          amount: '100',
+          asset_code: assetCode,
+          issuer_public_key: issuerPublicKey,
+          issuer_secret_key: issuerSecretKey,
+          receiver_public_key: receiverPublicKey,
+          receiver_secret_key: receiverSecretKey,
+        });
+
+        if ('transaction_hash' in response) {
+          expect(response.transaction_hash).toBeDefined();
+        } else {
+          throw new Error('Expected transaction_hash in response');
+        }
+      });
+
+      it('Should return an unsigned xdr when no receiver secret key is provided', async () => {
+        const response = await blindpay.stellar.mintAsset({
+          amount: '100',
+          asset_code: assetCode,
+          issuer_public_key: issuerPublicKey,
+          issuer_secret_key: issuerSecretKey,
+          receiver_public_key: receiverPublicKey,
+        });
+
+        if ('transaction_xdr' in response) {
+          expect(response.transaction_xdr).toBeDefined();
```

The same, the conditional is not necessary. If you can't build the builder, let it fail.